lib/libzfs/libzfs_config.c

```diff
@@ -341,7 +341,7 @@ check_restricted(const char *poolname)
 	static char *restricted = NULL;
 
 	const char *cur, *end;
-	int len, namelen;
+	uint32_t len, namelen;
```

Comment:
This doesn't look like it fixes the actual issue detected by Coverity.  And besides that this function is surprisingly hard to understand.  In a separate PR I'd suggest reworking it to use `strtok()`, similar to `zfs_resolve_shortname()`.